src/xrpld/rpc/detail/TransactionSign.cpp

```diff
@@ -710,6 +710,108 @@ transactionFormatResultImpl(Transaction::pointer tpTrans, unsigned apiVersion)
 
 //------------------------------------------------------------------------------
 
+static XRPAmount
+getTxFee(Application const& app, Config const& config, Json::Value tx)
+{
+    // autofilling only needed in this function so that the `STParsedJSONObject`
+    // parsing works properly it should not be modifying the actual `tx` object
+    if (!tx.isMember(jss::Fee))
+    {
+        tx[jss::Fee] = "0";
+    }
+
+    if (!tx.isMember(jss::Sequence))
+    {
+        tx[jss::Sequence] = "0";
+    }
+
+    if (!tx.isMember(jss::SigningPubKey))
+    {
+        tx[jss::SigningPubKey] = "";
+    }
+
+    if (!tx.isMember(jss::TxnSignature))
+    {
+        tx[jss::TxnSignature] = "";
+    }
+
+    if (tx.isMember(jss::Signers))
+    {
+        if (!tx[jss::Signers].isArray())
+            return config.FEES.reference_fee;
+
+        // check multi-signed signers
+        for (auto& signer : tx[jss::Signers])
+        {
+            if (!signer.isMember(jss::Signer) ||
+                !signer[jss::Signer].isObject())
+                return config.FEES.reference_fee;
+            if (!signer[jss::Signer].isMember(sfTxnSignature.jsonName))
+            {
+                // autofill TxnSignature
+                signer[jss::Signer][sfTxnSignature.jsonName] = "";
+            }
+        }
+    }
+
+    STParsedJSONObject parsed(std::string(jss::tx_json), tx);
+    if (!parsed.object.has_value())
+    {
+        return config.FEES.reference_fee;
+    }
+
+    try
+    {
+        STTx const& stTx = STTx(std::move(parsed.object.value()));
+        return calculateBaseFee(*app.openLedger().current(), stTx);
+    }
+    catch (std::exception& e)
+    {
+        return config.FEES.reference_fee;
+    }
+}
+
+Json::Value
+getCurrentNetworkFee(
+    Role const role,
+    Config const& config,
+    LoadFeeTrack const& feeTrack,
+    TxQ const& txQ,
+    Application const& app,
+    Json::Value& tx,
```

Comment:
Should be `Json::Value const& tx,`

Reply:
Fixed in a744645f7d2e7142512791a6bb2abaaff26bd62e